pkg/apis/maistra/v2/servicemeshcontrolplane_types.go

```diff
@@ -155,6 +154,11 @@ type ControlPlaneSpec struct {
 	// Runtime configuration for pilot (and galley, etc., pre 2.0)
 	// +optional
 	Runtime *ControlPlaneRuntimeConfig `json:"runtime,omitempty"`
+
+	ControlPlaneMode *ControlPlaneModeConfig `json:"controlPlaneMode,omitempty""`
+
+	RemoteMode *RemoteModeConfig `json:"remoteMode,omitempty"`
+
```

Comment:
I don't understand why we need two different settings.  Wouldn't this be easier to configure using an enum?  Is it possible to have both modes enabled?  Also, this might be better placed under the `General` section.

Reply:
+1

Also: please add a comment documenting what each of these fields is. The comments added here will end up being the documentation shown when users run `oc explain smcp.spec`, so their importance is beyond the scope of the code itself.

Reply:
@rcernich I think control and data plane parts will never want to co-exist in a single cluster (maybe in testing scenarios, but it is not a use-case for us), because that is the original deployment model which makes this change set completely unnecessary.
So do you suggest to implement a 3 way switch for this (`None`, meaning the original combined deployment, `ControlPlane` and `Remote`) similarly to how Telemetry works?

Reply:
@luksa any thoughts or concerns please?